The script correctly detect that is not from coreutils,
and fallback using another syntax which doesn't work
with BusyBox.
--- a/bin/elasticsearch-env
+++ b/bin/elasticsearch-env
@@ -88,6 +88,6 @@
   if [ $mktemp_coreutils -eq 0 ]; then
     ES_TMPDIR=`mktemp -d --tmpdir "elasticsearch.XXXXXXXX"`
   else
-    ES_TMPDIR=`mktemp -d -t elasticsearch`
+    ES_TMPDIR=`mktemp -d -t elasticsearch.XXXXXXXX`
   fi
 fi
